### ISSUE TITLE
Bug: `AttributeError: 'wrapper_descriptor' object has no attribute '__module__'`

### ISSUE THREAD

**Issue by lqmanh** [litestar-org/litestar#1504]
### Description

Litestar app fails to start, due to the above error.

### URL to code causing the issue

https://github.com/litestar-org/litestar/blob/4f734ea5970d414d49b92cf3c49988aa158cad30/litestar/utils/signature.py#L64

### MCVE

```python
from uuid import UUID

from litestar import Controller, Litestar, get
from litestar.di import Provide
from pydantic import BaseModel


class TestService:
    ...


class TestOutput(BaseModel):
    id: UUID


class TestController(Controller):
    path = "/test"
    dependencies = {
        "test_service": Provide(TestService),
    }

    @get("/{test_id:uuid}")
    async def get_by_id(self, test_id: UUID) -> TestOutput:
        return TestOutput(id=test_id)


app = Litestar(
    route_handlers=[TestController],
)
```

### Steps to reproduce

Run the app in MCVE section above.

### Logs

```bash
File "/app/src/asgi.py", line 11, in <module>
    app = Litestar(
          ^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litestar/app.py", line 438, in __init__
    self.register(route_handler)
  File "/usr/local/lib/python3.11/site-packages/litestar/app.py", line 546, in register
    self._create_handler_signature_model(route_handler=route_handler)
  File "/usr/local/lib/python3.11/site-packages/litestar/app.py", line 775, in _create_handler_signature_model
    parsed_signature=ParsedSignature.from_fn(
                     ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litestar/utils/signature.py", line 283, in from_fn
    fn_type_hints = get_fn_type_hints(fn, namespace=signature_namespace)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litestar/utils/signature.py", line 64, in get_fn_type_hints
    **vars(sys.modules[fn_to_inspect.__module__]),
                       ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'wrapper_descriptor' object has no attribute '__module__'. Did you mean: '__reduce__'?
```

### Litestar Version

- `main` branch
- v2.0.0alpha4

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

**Comment by peterschutt** [comment]
Thanks for the report @lqmanh - if possible, can you provide an example of the code that reproduces this?

**Comment by lqmanh** [comment]
@peterschutt I've updated the MCVE code

**Comment by peterschutt** [comment]
Thanks @lqmanh - fix in #1505
